### PR TITLE
Add tags variable to allow passing in extra tags (copied from terraform-aws-consul)

### PR DESCRIPTION
**Pull request by Tethik**
Basically a shameless copy of https://github.com/hashicorp/terraform-aws-consul/pull/47

Tried to run the tests, but https://github.com/gruntwork-io/terratest gives a 404.

I tested it on my own work-in-progress cluster with and without the variable. Both worked.

### CODE REVIEW

**Comment by Tethik** [comment]
Maybe #9 should be addressed at the same time.

File: modules/nomad-cluster/main.tf [hunk 1]
```diff
@@ -28,11 +28,19 @@ resource "aws_autoscaling_group" "autoscaling_group" {
...
-    key                 = "${var.cluster_tag_key}"
-    value               = "${var.cluster_name}"
-    propagate_at_launch = true
-  }
+  tags = [
+    {
+      key                 = "Name"
+      value               = "${var.cluster_name}"
+      propagate_at_launch = true
+    },
```
Comment (brikis98) [inline modules/nomad-cluster/main.tf]: I'm guessing this added `Name` tag is a copy/paste error from the other PR? The `cluster_tag_key` is set to `Name` by default.

Comment (Tethik) [reply modules/nomad-cluster/main.tf]: Yep, shameless copypasta is even more shameless :fearful: 

I'll commit a fix and address #9 at the same time.

**Comment by Tethik** [comment]
I committed the changes to make this module the same as terraform-aws-consul now (#9), but it will make potentially make some breaking changes due to how the tagging is done.

I added the new variable `cluster_tag_value`. By default it has the same value as in the consul-module (defaults to auto-join), this will conflict with the current tag value of `var.cluster_name` (nomad-example in the example).

So I think people upgrading to a new version with this PR will need to make two changes:
1. Add the `cluster_tag_value` variable to their module usage.
2. Ensure `cluster_tag_value` is set to the same as `cluster_name` to ensure new nodes join the same cluster. 

Is this ok or is there a different approach we should try to fix this?

**Comment by brikis98** [comment]
Thanks! Although backwards incompatible, this is more consistent with consul, so I think it's a good approach.

**Comment by brikis98** [comment]
https://github.com/hashicorp/terraform-aws-nomad/releases/tag/v0.2.0

**Pull request merged**